sktime/transformations/series/bootstrapping.py

```diff
@@ -0,0 +1,401 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Bootstrapping methods for time series."""
+
+__author__ = ["ltsaprounis"]
+
+from copy import copy
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from statsmodels.tsa.api import STL as _STL
+
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+class BootsrappingTransformer(BaseTransformer):
+    """Creates a population of similar time series.
+
+    This method utilises a form of bootstrapping to generate a population of
+    similar time series to the input time series [1]_, [2]_.
+
+    First the observed time series is transformed using a Box-Cox transformation to
+    stabilise the variance. Then it's decomposed to seasonal, trend and residual
+    time series, using the STL implementation from statsmodels [4]_. We then sample
+    blocks from the residuals time series using the Moving Block Bootstrapping (MBB)
+    method [3]_ to create synthetic residuals series that mimic the autocorrelation
+    patterns of the observed series. Finally these bootstrapped residuals are added
+    to the season and trend components and we use the inverse Box-Cox transform to
+    return a panel of similar time series.
+
+    The resulting panel can be used for Bagging forecasts, prediction intervals and
+    data augmentation.
+
+    Parameters
+    ----------
+    n_series : int, optional
+        The number of bootstraped time series that will be generated, by default 10
+    sp : int, optional
+        Seasonal periodicity of the data in integer form, by default 12.
+        Must be an integer >= 2
+    block_length : int, optional
+        The length of the block in the MBB method, by default None.
+        If not provided, the following heuristic is used, the block length will the
+        minimum between 2*sp and len(X) - sp.
+    sampling_replacement: bool, optional
+        Whether the MBB sample is with or without replacement, by default False.
+    return_actual : bool, optional
+        If True the output will contain the actual time series, by default True.
+        The actual time series will be labelled as "<series_name>_actual" (or "actual"
+        if series name is None).
+    series_name : str, optional
```

Comment:
should this not be passed as the column name in a single column `pd.DataFrame`, or name of a `pd.Series`, instead of a parameter?